sources/packages/backend/apps/queue-consumers/src/services/cas-invoice/cas-invoice.service.ts

```diff
@@ -0,0 +1,264 @@
+import { Injectable } from "@nestjs/common";
+import {
+  CASService,
+  PendingInvoicePayload,
+  InvoiceLineDetail,
+  SendPendingInvoicesResponse,
+} from "@sims/integrations/cas";
+import { CAS_BAD_REQUEST } from "@sims/integrations/constants";
+import {
+  CASInvoice,
+  CASInvoiceBatchApprovalStatus,
+  CASInvoiceStatus,
+  User,
+} from "@sims/sims-db";
+import {
+  processInParallel,
+  CustomNamedError,
+  getAbbreviatedDateOnlyFormat,
+} from "@sims/utilities";
+import { ProcessSummary } from "@sims/utilities/logger";
+import { DataSource, Repository } from "typeorm";
+import { InjectRepository } from "@nestjs/typeorm";
+import { SystemUsersService } from "@sims/services";
+
+@Injectable()
+export class CASInvoiceService {
+  constructor(
+    private readonly systemUsersService: SystemUsersService,
+    private readonly casService: CASService,
+    @InjectRepository(CASInvoice)
+    private readonly casInvoiceRepo: Repository<CASInvoice>,
+    private readonly dataSource: DataSource,
+  ) {}
+  /**
+   * Get the list of pending invoices to be sent from the approved batch.
+   * @returns list of pending invoices.
+   */
+  async getPendingInvoices(): Promise<CASInvoice[]> {
+    const pendingInvoices = await this.dataSource
+      .getRepository(CASInvoice)
+      .find({
+        select: {
+          id: true,
+          casSupplier: {
+            id: true,
+            supplierNumber: true,
+            supplierAddress: { supplierSiteCode: true },
```

Comment:
`supplierAddress` is a JSONB and unless something changed on typeorm this code will retrieve all data and using like this can be misleading.
Please execute it as `supplierAddress: true as unknown`.
In case some clarification is needed, please search the code by ` true as unknown` to see similar places doing the same.